library/src/camera_impl.cxx

```diff
@@ -0,0 +1,241 @@
+  vtkRenderer* VTKRenderer = nullptr;
+  vtkNew<vtkCamera> DefaultVTKCamera;
+};
+
+//----------------------------------------------------------------------------
+camera_impl::camera_impl()
+ : Internals(new camera_impl::internals()){}
+
+//----------------------------------------------------------------------------
+camera_impl::~camera_impl() = default;
+
+//----------------------------------------------------------------------------
+void camera_impl::setPosition(const vector_t& pos)
+{
+  vtkCamera* cam = this->GetVTKCamera();
+  cam->SetPosition(pos.data());
+  cam->OrthogonalizeViewUp();
+  this->Internals->VTKRenderer->ResetCameraClippingRange();
+}
+
+//----------------------------------------------------------------------------
+camera::vector_t camera_impl::getPosition()
+{
+  vector_t pos;
+  vtkCamera* cam = this->GetVTKCamera();
+  cam->GetPosition(pos.data());
+  return pos;
+}
+
+//----------------------------------------------------------------------------
+void camera_impl::setFocalPoint(const vector_t& foc)
+{
+  vtkCamera* cam = this->GetVTKCamera();
+  cam->SetFocalPoint(foc.data());
+  cam->OrthogonalizeViewUp();
+  this->Internals->VTKRenderer->ResetCameraClippingRange();
+}
+
+//----------------------------------------------------------------------------
+camera::vector_t camera_impl::getFocalPoint()
+{
+  vector_t foc;
+  vtkCamera* cam = this->GetVTKCamera();
+  cam->GetFocalPoint(foc.data());
+  return foc;
+}
+
+//----------------------------------------------------------------------------
+void camera_impl::setViewUp(const vector_t& up)
+{
+  vtkCamera* cam = this->GetVTKCamera();
+  cam->SetViewUp(up.data());
+  cam->OrthogonalizeViewUp();
+  this->Internals->VTKRenderer->ResetCameraClippingRange();
+}
+
+//----------------------------------------------------------------------------
+camera::vector_t camera_impl::getViewUp()
+{
+  vector_t up;
+  vtkCamera* cam = this->GetVTKCamera();
+  cam->GetViewUp(up.data());
+  return up;
+}
+
+//----------------------------------------------------------------------------
+void camera_impl::setViewAngle(const double& angle)
+{
+  vtkCamera* cam = this->GetVTKCamera();
+  cam->SetViewAngle(angle);
+  cam->OrthogonalizeViewUp();
+  this->Internals->VTKRenderer->ResetCameraClippingRange();
+}
+
+//----------------------------------------------------------------------------
+double camera_impl::getViewAngle()
+{
+  vtkCamera* cam = this->GetVTKCamera();
+  return cam->GetViewAngle();
+}
+
+//----------------------------------------------------------------------------
+void camera_impl::setViewMatrix(const matrix_t& matrix)
+{
+  vtkCamera* cam = this->GetVTKCamera();
+  cam->SetPosition(matrix[12], matrix[13], matrix[14]);
+  cam->SetFocalPoint(matrix[8] + matrix[12], matrix[9] + matrix[13], matrix[10] + matrix[14]);
+  cam->SetViewUp(matrix[4], matrix[5], matrix[6]);
+//   cam->OrthogonalizeViewUp(); TODO NOT NEEDED ?
+  this->Internals->VTKRenderer->ResetCameraClippingRange();
```

Comment:
does that looks correct to you @Meakk ?

Reply:
Yes, you need to reset the clipping range everytime the camera position/orientation changes

Reply:
my question is about the whole method

Reply:
Yes, should be good. And indeed `cam->OrthogonalizeViewUp()` is not needed.